service/history/queues/executable.go

```diff
@@ -334,62 +312,112 @@ func (e *executableImpl) HandleErr(err error) (retErr error) {
 			}
 			e.resourceExhaustedCount++
 			metrics.TaskThrottledCounter.With(e.taggedMetricsHandler).Record(1)
-			return err
+			return true, err
 		}
 
 		err = consts.ErrResourceExhaustedBusyWorkflow
 	}
 	e.resourceExhaustedCount = 0
 
 	if _, isNotFound := err.(*serviceerror.NotFound); isNotFound {
-		return nil
+		return true, nil
 	}
 
 	// This means that namespace is deleted, and it is safe to drop the task (=ignore the error).
 	if _, isNotFound := err.(*serviceerror.NamespaceNotFound); isNotFound {
-		return nil
+		return true, nil
+	}
+
+	if _, isDataLoss := err.(*serviceerror.DataLoss); isDataLoss {
+		e.logger.Error("Dropping task due to data loss", tag.Error(err))
+		return true, nil
```

Comment:
I think this should be treated the same as those terminal serialization errors. and put to DLQ if enabled.